apis/python/src/tiledbsoma/_scene.py

```diff
@@ -245,21 +247,66 @@ def add_new_multiscale_image(
             Experimental.
         """
         if transform is not None:
-            raise NotImplementedError()
+            # Get and check the scene coordinate space axis names.
+            if self.coordinate_space is None:
+                raise SOMAError(
+                    "The scene coordinate space must be set before setting a transform."
+                )
+            if transform.input_axes != self.coordinate_space.axis_names:
+                raise ValueError(
+                    f"The name of the transform input axes, {transform.input_axes}, "
+                    f"do not match the name of the axes, "
+                    f"{self.coordinate_space.axis_names}, in the scene coordinate "
+                    f"space."
+                )
+
+            # Get and check the multiscale image coordinata space axis names.
+            # Note: The input paremeters to the MultiscaleImage create method are being
+            #   revisited. The following code will be improved after the create
+            #   parameters stabilize.
+            ordered_axis_names: List[Optional[str]] = [None, None, None]
+            for ax_name, ax_type in zip(axis_names, axis_types):
+                # Validation unneed if the type falls through. Invalid types will be
+                # caught in the MultiscaleImage.create method.
+                if ax_type == "width":
+                    ordered_axis_names[0] = ax_name
+                elif ax_type == "height":
+                    ordered_axis_names[1] = ax_name
+                elif ax_type == "depth":
+                    ordered_axis_names[2] = ax_name
+            ordered_axis_names = [
+                axis_name for axis_name in ordered_axis_names if axis_name is not None
+            ]
+            if transform.output_axes != tuple(ordered_axis_names):
+                raise ValueError(
+                    f"The name of the transform output axes, {transform.output_axes}, "
+                    f"do not match the name of the axes, {tuple(ordered_axis_names)}, "
+                    f"of the coordinate space the multiscale image is defined on."
+                )
 
+        # Open the subcollection and add the new multiscale image.
         coll = self._open_subcollection(subcollection)
-        return coll._add_new_element(
+        image = coll._add_new_element(
             key,
             MultiscaleImage,
             lambda create_uri: MultiscaleImage.create(
                 create_uri,
                 context=self.context,
                 tiledb_timestamp=self.tiledb_timestamp_ms,
+                axis_names=axis_names,
+                axis_types=axis_types,
                 **kwargs,
             ),
             uri,
         )
 
+        # Store the metadata for the transform.
+        if transform is not None:
+            coll.metadata[f"soma_scene_registry_{key}"] = transform_to_json(transform)
```

Comment:
Not in this PR, but one thing I'd like to do in the future is make it so that the user can't modify these metadata values directly.

In the C++ API, we [throw an error](https://github.com/single-cell-data/TileDB-SOMA/blob/main/libtiledbsoma/src/soma/soma_array.cc#L1090-L1094) if there is an attempt to modify these reserved metadata keys. BUT there is a way to [bypass](https://github.com/single-cell-data/TileDB-SOMA/blob/main/libtiledbsoma/src/soma/soma_array.cc#L1089) this by passing the `force` flag. Internally, we could use `coll._handle.add_metadata(f"soma_scene_registry_{key}", transform_to_json(transform), force=True)`. This way we can still modify the metadata values as needed but the user cannot.